docs/docsite/rst/lookup_guide.rst

```diff
@@ -0,0 +1,159 @@
+.. _ansible_collections.community.hashi_vault.docsite.lookup_guide:
+
+************
+Lookup guide
+************
+
+This guide is not a comprehensive listing of included lookup plugins and how to use them, rather it is intended to explain the role of the lookup plugins in ``community.hashi_vault`` and how they are they used, especially when compared to modules of the same name.
+
+.. contents::
+  :local:
+  :depth: 2
+
+
+About the ``hashi_vault`` lookup
+================================
+
+The ``hashi_vault`` lookup plugin is the oldest Vault-related content in Ansible. It was included in pre-collections Ansible (<2.10). As a result, it's the most used plugin for Vault, and the one most people are familiar with.
+
+``hashi_vault`` lookup considerations
+-------------------------------------
+
+Due to the history of the ``hashi_vault`` lookup plugin, it does many jobs. It is versatile, but sometimes unintuitive.
+
+The ``hashi_vault`` lookup plugin performs three main tasks:
+
+ - authentication, taking parameters for various login types, performing a login, and acquiring a token with which it can make additional calls to Vault.
+- a generic read operation, which allows it to read any kind of Vault path, without having to be written with that type of path in mind.
+- reading secrets.
+
+Reading secrets is the most common use case, with the ``kv`` (key/value) store built into Vault as by far the most common secret store. Most implementations use v2 of the ``kv`` store. To make reading v2 ``kv`` secrets easy, the lookup plugin assumes that you're probably trying to read a ``kv`` secret, and tries to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrapped in another structure. The lookup plugin seeks to make ``kv2`` responses look more like responses from version 1.
```

Comment:
```suggestion
 - authentication, taking parameters for various login types, performing a login, and acquiring a token with which it can make additional calls to Vault.
- a generic read operation, which allows it to read any kind of Vault path, without having to be written with that type of path in mind.
- modifying the response from Vault to be more useful it came from a ``kv2`` backend.

Expanding on that last point: reading secrets is the most common use case, with the ``kv`` (key/value) store built into Vault as by far the most common secret store. Most implementations use v2 of the ``kv`` store. To make reading v2 ``kv`` secrets easy, the lookup plugin assumes that you're probably trying to read a ``kv`` secret, and tries to infer if the response is from ``kv2``, because the responses from version 2 include metadata and have the secret value additionally wrapped in another structure. In short: the ``hashi_vault`` lookup seeks to make ``kv2`` responses look more like the (simpler) responses from version 1.
```

Reply:
@acozine I committed your previous suggestion and then made a few changes onto it, lmk what you think.

Reply:
Nice!